### ISSUE TITLE
Poor UI Performance

### ISSUE THREAD

**Issue by KakarottoCake** [jellyfin/jellyfin-media-player#299]
The performance of the UI of the Jellyfin Media Player is noticeably laggy compared to the UI of the web UI of the Web player.

 - OS: Windows 11
 - Windows Update 21H2
**Please note, this has nothing to do with the video playback of the application. Only the UI**
https://user-images.githubusercontent.com/61133727/187061149-7997520d-f218-4d5d-a0d0-ad9cef0e45a4.mp4

**Comment by flo333** [comment]
Haha, I thought it was just me. Noticed the same on Linux with 1.7.1 installed from the AUR.

**Comment by KakarottoCake** [comment]
@flo333 it's kinda sad they haven't addressed this yet. Such a nice app that's being held back by it's performance

**Comment by flo333** [comment]
@KakarottoCake Yeah. I ended up using [jellyfin-mpv-shim](https://github.com/jellyfin/jellyfin-mpv-shim) and the WebUI as a controller. Much smoother experience.

**Comment by iwalton3** [comment]
I am aware that there are performance issues with the web UI. Some of the enhancements I have done in the past include enabling hardware accelerated rendering of the webengine and keeping QT 5.x up to date.

There is a Qt6 branch that could help with this, but it is currently affected by s substantial bug: https://bugreports.qt.io/browse/QTBUG-111739

**Comment by KakarottoCake** [comment]
> Any updates or fixes we can do here? The performance when editing my library etc. on the Jellyfin desktop app on Windows 10 has gotten literally unusable, but is fine on the web app.

I can vouch, it has only gotten worse. Sadly I think this is very low priority for the Dev team...

**Comment by thornbill** [comment]
Well the "dev team" for JMP is mainly a single person who has had a lot going on in their personal life so let's have some patience here.

**Comment by KakarottoCake** [comment]
> Well the "dev team" for JMP is mainly a single person who has had a lot going on in their personal life so let's have some patience here.

I'm not sure how I'm supposed to respond to that. Sorry for what's happening I guess 😕. I did want to point out the issue was made in 2022 originally so patience is really not an issue here. Just an update would be nice

**Comment by thornbill** [comment]
When there is an update to provide I'm sure it will be shared here. In a 100% volunteer driven open source project an issue being open for a long time like this is really a failure of the community to step up and help resolve issues that they feel strongly about or are seriously impacted by.

**Comment by iwalton3** [comment]
Contributions are welcome and highly encouraged. I have had a lot in my personal life come up over the past year.

This project is something I care about and I watch the Jellyfin internal chats for critical issues coming up. The application still mostly works but I need to set aside probably an entire weekend just to dig through the issues and update all the dependencies. I haven't gotten a chance yet since I have been underwater with life stuff.

I'm aware of this laggy UI issue but I don't know any way to fix it. It is a problem which also plagued Plex Media Player too. Help would be greatly appreciated by anyone who has any ideas.

**Comment by SnowSquire** [comment]
I was having this issue and reinstalled windows and still have it might be a hardware things. Specs

CPU: AMD 7950X3d
GPU: RTX 3060TI 
Windows 11 23H2

What system do you test on @iwalton3?

**Comment by iwalton3** [comment]
Debian Stable is what I test on

**Comment by GRFTSOL** [comment]
I have same problem, but I found that if turn up play speed, the fps of player ui is also multiplied, e.g.: If I'm playing a 25fps video, player ui will limited to 25fps, when I use 2x speed, its limit will be 50fps 

And when I pause the video, it turns back to 60fps, can tell from the video.

Found is via RTSS because I forgot to turn off it and I was in 1.5x speed playing some episode, RTSS shows in the corner.

@iwalton3 If this would help with identifying the problem?

https://github.com/user-attachments/assets/2c9603f2-ccb6-47e2-9e77-141f8f1c4801

**Comment by GRFTSOL** [comment]
> I have same problem, but I found that if turn up play speed, the fps of player ui is also multiplied, e.g.: If I'm playing a 25fps video, player ui will limited to 25fps, when I use 2x speed, its limit will be 50fps
> 
> And when I pause the video, it turns back to 60fps, can tell from the video.
> 
> Found is via RTSS because I forgot to turn off it and I was in 1.5x speed playing some episode, RTSS shows in the corner.
> 
> @iwalton3 If this would help with identifying the problem?
> 2024-10-10.12-15-26.mp4

I tried this client setting to improve for now, hope this can be fix someday.
![图片](https://github.com/user-attachments/assets/1b998c5c-b22a-4052-9165-90eb2e489b15)
with deinterlace active, the playing interface can reach 2x the frame rate of the video, see below, 25fps video with 50 fps interface, better than before.
![图片](https://github.com/user-attachments/assets/4db02089-879d-4366-b382-20d381fb23d6)

**Comment by casualkitten1** [comment]
@GRFTSOL 's comment fixed it on my end.

The UI was a bit slow/laggy (would . Using the deinterlace function fixed it.

Thanks.
